### ISSUE TITLE
Simulate styles for extended form elements

### ISSUE THREAD

**Issue by nazar-pc** [Polymer/polymer#2337]
``` html
<dom-module id="cs-textarea">
  <link href="style.css" rel="import" type="css">
  <script>
    Polymer({
      'is': 'cs-textarea',
      'extends': 'textarea'
    });
  </script>
</dom-module>
```

Styles can't be applied to `textarea` in Chromium because it uses native Shadow DOM that can't be modified correctly at the moment (I'm using `Polymer.dom = 'shadow'`), but nothing prohibits so simulate it like in browsers that doesn't have native Shadow DOM support, for instance, it works nicely in Firefox.

The same story about `input`, we can't do that directly, but we can simulate it, and simulation works nicely in Firefox.

So basically, there is a need to process styles for extended `textarea`, `input` (probably some other elements) regardless native Shadow DOM support like for browsers that lacks native support.

**Comment by sorvell** [comment]
This is something we're considering for the next major version of Polymer. To support a use case like this, an element needs to be able to identify its styling scope (document or a containing shadowRoot) and put styles there.

**Comment by nazar-pc** [comment]
I've worked on this today, but the main obstacle is that code relies heavily on global `Polymer.Settings.useNativeShadow`, while in order for this to work we basically need similar per-element property like `element._supportsNativeShadowRootCreation` to be used instead.

With per-element property we can force needed elements that doesn't support Shadow Root creation on them (https://github.com/w3c/webcomponents/issues/110, https://w3c.github.io/webcomponents/spec/shadow/#methods, but Chromium additionally supports currently `<button>` at least) to use the same mechanism as browsers without native Shadow DOM support (uses full Shadow DOM polyfill).

**Comment by nazar-pc** [comment]
Here is the simple test case that should pass when implemented: https://github.com/Polymer/polymer/compare/master...nazar-pc:type-extension-test-case

**Comment by TimvdLippe** [comment]
Since extending native elements is discouraged (see https://www.polymer-project.org/2.0/docs/upgrade#remove-type-extension-elements for a larger explanation), you are now required to wrap the native element. As such, any styling concerns fall naturally into place where you can just style the native element directly. Therefore I am closing this issue for now.
